### PR TITLE
feat: implement 'answer a question logic'

### PR DESCRIPTION
**Pull request by DjamelALI**
- [x]  Add a prisma answer table
- [x]  Add a answer module
- [ ]  Add a graphql query and mutations
    - [x]  query by id
    - [x]  query list by user
    - [x]  mutation create answer
    - [ ]  mutation edit answer saving the previous version
    - [x]  mutation delete answer

### CODE REVIEW

File: back/prisma/schema.prisma [hunk 1]
```diff
@@ -36,6 +37,20 @@ model Question {
   body         String
   files        File[]
   user         User     @relation(fields: [userId], references: [id])
+  Answer       Answer[]
```
Comment (roadroot) [inline back/prisma/schema.prisma:40]: Please remove this line
```suggestion
```

File: back/src/answer/answer.model.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { UserDTO } from 'src/user/model/user.model';
+import { Field, Int, ObjectType } from '@nestjs/graphql';
+
+@ObjectType('Answer')
+export class AnswerDTO {
+  @Field()
+  id: string;
+
+  @Field(() => UserDTO, { nullable: true })
```
Comment (roadroot) [inline back/src/answer/answer.model.ts:9]: Type is not required
```suggestion
  @Field({ nullable: true })
```

File: back/src/answer/answer.model.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  userId: string;
+
+  updateDate: Date;
+
+  creationDate: Date;
+
+  @Field({nullable: true})
+  title?: string;
+
```
Comment (roadroot) [inline back/src/answer/answer.model.ts:20]: Please add files

File: back/src/answer/answer.module.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { Module, forwardRef } from '@nestjs/common';
+import { AnswerService } from './answer.service';
+import AnswerResolver from './answer.resolver';
+import { UserModule } from 'src/user/user.module';
+import { PrismaModule } from 'src/prisma/prisma.module';
+
+@Module({
+  imports: [forwardRef(() => UserModule), PrismaModule],
```
Comment (roadroot) [inline back/src/answer/answer.module.ts:8]: Why did you use forwardRed?

File: back/src/app.module.ts [hunk 5]
```diff
@@ -9,17 +9,19 @@ import { QuestionModule } from './question/question.module';
...
+import { AnswerModule } from './answer/answer.module';
 import { ApolloServerPlugin } from 'apollo-server-plugin-base';
 
+
 const myPlugin: ApolloServerPlugin = {
   // Fires whenever a GraphQL request is received from a client.
   async requestDidStart(requestContext) {
     console.log('Request started! Query:\n' + requestContext.request.query);
-
+    const c = requestContext.debug ?? false;
```
Comment (roadroot) [inline back/src/app.module.ts:20]:
```suggestion
```
Remove this line as it is useless

File: back/src/app.module.ts [hunk 6]
```diff
@@ -9,17 +9,19 @@ import { QuestionModule } from './question/question.module';
...
     return {
       async didEncounterErrors(requestContext) {
         console.log(
-          `Encountered errors! ${JSON.stringify(requestContext.errors)}`,
+          `Encountered errors! ${requestContext.errors[0]}`,
```
Comment (roadroot) [inline back/src/app.module.ts:24]:
```suggestion
          `Encountered errors! ${requestContext.errors}`,
```
keep previous syntax

**Review by roadroot — CHANGES_REQUESTED** [review]

**Comment by roadroot** [comment]
Please resolve the field answers in UserDTO to be able to query all the answers of a specific user

File: back/src/answer/answer.model.ts (hunk 3 above)
Comment (DjamelALI) [reply back/src/answer/answer.model.ts:22]: Done

File: back/src/answer/answer.module.ts (hunk 4 above)
Comment (DjamelALI) [reply back/src/answer/answer.module.ts]: I removed it, I had at some time a circular dependency, ty!

File: back/src/app.module.ts (hunk 6 above)
Comment (DjamelALI) [reply back/src/app.module.ts]: Done

File: back/src/answer/answer.resolver.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    @Args('data') data: AnswerDataDTO,
+    @CurrentUser() user: UserDTO,
+  ): Promise<AnswerDTO> {
+    return await this.answer.createAnswer({ ...data, userId: user.id });
+  }
+
+  @UseGuards(JwtAuthGuard)
+  @Mutation(() => AnswerDTO)
+  async delete(
+    // TODO add current user
```
Comment (roadroot) [inline back/src/answer/answer.resolver.ts:48]: Please remove this TODO as it is done
```suggestion
```

**Comment by roadroot** [comment]
> Owner

Done

**Review by roadroot — APPROVED** [review]

**Pull request merged**